src/main/js/providers/ProviderTable.js

```diff
@@ -59,16 +58,18 @@ class ProviderTable extends Component {
     }
 
     combineModelWithDefaults(providerConfig) {
-        const { descriptor } = this.state;
-        if (!descriptor) {
-            return {};
-        }
-        const emptyConfig = FieldModelUtilities.createFieldModelWithDefaults(descriptor.fields, DescriptorUtilities.CONTEXT_TYPE.GLOBAL, descriptor.name);
-        const updatedFieldModel = FieldModelUtilities.combineFieldModels(emptyConfig, providerConfig);
-        if (providerConfig.id) {
-            updatedFieldModel.id = providerConfig.id;
+        const { descriptors, descriptorName } = this.props
+        const descriptorList = DescriptorUtilities.findDescriptorByNameAndContext(descriptors, descriptorName, DescriptorUtilities.CONTEXT_TYPE.GLOBAL)
+        if (descriptorList && descriptorList.length > 0) {
+            const foundDescriptor = descriptorList[0];
+            const emptyConfig = FieldModelUtilities.createFieldModelWithDefaults(foundDescriptor.fields, DescriptorUtilities.CONTEXT_TYPE.GLOBAL, foundDescriptor.name);
+            const updatedFieldModel = FieldModelUtilities.combineFieldModels(emptyConfig, providerConfig);
+            if (providerConfig.id) {
+                updatedFieldModel.id = providerConfig.id;
+            }
+            return updatedFieldModel;
         }
-        return updatedFieldModel;
+        return {};
```

Comment:
Should this return an empty object? from the function name, I imagine this should return at least the original model without defaults.

Reply:
Previously we had 
``` java
 if (!descriptor) {
            return {};
        }
```
So this maintains that behavior.